flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/model/SourceBuilder.java

```diff
@@ -602,38 +602,64 @@ private void addImports() {
         Range startLineRange = CommonUtils.toRange(startLine);
 
         // Obtain the module descriptor of the current module
-        String currentModuleOrg;
-        String currentModuleName;
         Optional<Module> currentModule = this.workspaceManager.module(filePath);
-        boolean isGenerated;
+
+        // Clean the imports for the current module if the imports belong to the same package
         if (currentModule.isPresent()) {
             ModuleDescriptor descriptor = currentModule.get().descriptor();
-            currentModuleName = descriptor.name().toString();
-            currentModuleOrg = descriptor.org().value();
-            imports.remove(currentModuleOrg + "/" + currentModuleName);
-            isGenerated = Boolean.TRUE.equals(flowNode.codedata().isGenerated());
-        } else {
-            currentModuleName = "";
-            isGenerated = false;
+            String currentModuleName = descriptor.name().toString();
+            String currentModuleOrg = descriptor.org().value();
+            cleanImportsForModuleImportsInSamePackage(currentModuleOrg, currentModuleName);
         }
 
         // Remove the existing imports
+        removeExistingImports(syntaxTree);
+
+        // Generate the text edits for the imports
+        generateImportsTextEdits(filePath, startLineRange);
+    }
+
+    private void cleanImportsForModuleImportsInSamePackage(String orgName, String moduleName) {
+        for (String importStmt : imports) {
+            String[] parts = importStmt.split("/");
+            if (parts.length > 1) {
+                String importOrg = parts[0];
+                String importModule = parts[1];
+
+                if (!orgName.equals(importOrg)) {
+                    continue;
+                }
+
+                if (moduleName.equals(importModule)) {
+                    imports.remove(orgName + "/" + moduleName);
```

Comment:
Without modifying the iterating set, shall we modify a duplicated set?

Reply:
Good catch. Will resolved this.

Reply:
Fixed in 12811e61f179d88603cc5b4317f3fc6035dbdadd